libs/vkd3d/command.c

```diff
@@ -1508,6 +1509,43 @@ static ULONG STDMETHODCALLTYPE d3d12_shared_fence_AddRef(d3d12_fence_iface *ifac
     return refcount;
 }
 
+static void d3d12_shared_fence_open_kmt(struct d3d12_shared_fence *fence, struct d3d12_device *device)
+{
+#ifdef _WIN32
+    const struct vkd3d_vk_device_procs *vk_procs = &device->vk_procs;
+    VkSemaphoreGetWin32HandleInfoKHR win32_handle_info;
+    D3DKMT_OPENSYNCOBJECTFROMNTHANDLE open = {0};
+    NTSTATUS status;
+    VkResult vr;
+
+    win32_handle_info.sType = VK_STRUCTURE_TYPE_SEMAPHORE_GET_WIN32_HANDLE_INFO_KHR;
+    win32_handle_info.pNext = NULL;
+    win32_handle_info.semaphore = fence->timeline_semaphore;
+    win32_handle_info.handleType = VK_EXTERNAL_SEMAPHORE_HANDLE_TYPE_D3D12_FENCE_BIT;
+
+    if ((vr = VK_CALL(vkGetSemaphoreWin32HandleKHR(device->vk_device, &win32_handle_info, &open.hNtHandle))))
```

Comment:
I have similar concerns as for memory objects. It looks as-if we're importing, but this is actually export? Maybe shared_fence_open_kmt name should be clarified so it's obvious it is exporting.

Reply:
Same thing as above, this is to open a local KMT handle, which are necessary to create the exported handles in CreateSharedHandle with D3DKMTShareObjects.